fabric/bin/peer.sh

```diff
@@ -17,298 +29,211 @@ FABRIC_SCRIPTDIR="${FPC_TOP_DIR}/fabric/bin/"
+	    esac
     done
-    # accept only if original install is of same "type" (fpc vs vanilla)
-    $(is_fpc_cc ${CC_NAME} ${CC_VERSION})
-    FPC_INSTALLED=$?
-    [ "${CC_LANG}" = "fpc-c" ];
-    FPC_UPGRADE=$?
-    if [ ${FPC_INSTALLED} != ${FPC_UPGRADE} ]; then
-	die "cannot change frpm FPC to non-FPC or vice-versa in chaincode upgrade"
+    if [ ! "${CC_LANG}" = "fpc-c" ]; then
+	# Nothing special to do for non-fpc chaincode, just forward to real peer
+	return
     fi
 
-    # if fpc, do same switcheroo and docker-build as done by install
-    if [ "${CC_LANG}" = "fpc-c" ]; then                                                                                                   if [ -z ${CC_NAME+x} ] || [ -z ${CC_VERSION+x} ] || [ -z ${CC_ENCLAVESOPATH+x} ] ; then
-            # missing params, don't do anything and let real peer report errors
-            return
-        fi
-        yell "Found valid FPC Chaincode Install!"
-        # get net-id and peer-id from core.yaml, for now just use defaults ...
-        parse_fabric_config .
-
-        # add private chaincode name prefix
-        FPC_NAME=$(get_fpc_name ${CC_NAME})
-
-        # get docker name for this chaincode
-        DOCKER_IMAGE_NAME=$(${FPC_DOCKER_NAME_CMD} --cc-name ${FPC_NAME} --cc-version ${CC_VERSION} --net-id ${NET_ID} --peer-id ${PEER_ID}) || die "could not get docker image name"
-
-        # install docker
-        try make ENCLAVE_SO_PATH=${CC_ENCLAVESOPATH} DOCKER_IMAGE=${DOCKER_IMAGE_NAME} -C ${FPC_TOP_DIR}/ecc docker-fpc-app
-
-        # replace path and lang arg with dummy go chaincode
-        ARGS_EXEC=( 'chaincode' 'upgrade' '-n' "${FPC_NAME}" '-v' "${CC_VERSION}" '-p' 'github.com/hyperledger/fabric/examples/cha
-incode/go/example02/cmd' "${OTHER_ARGS[@]}" )
-    fi
-    return
+    # check required parameters
+    [ ! -z "${OUTPUTFILE}" ]     || die "no or ill-defined outputfile provided"
+    [ ! -z "${CC_LABEL}" ]       || die "undefined parameter '--label'"
+    [ -d "${CC_ENCLAVESOPATH}" ] || die "undefined or non-existing '-p'/'--path' parameter '${CC_ENCLAVESOPATH}'"
+
+    # Note: normal fabric package format & layout:
+    # Overall the package is a gzipped tar-file containing files
+    # - 'metadata.json', a json object with 'path', 'type' and 'label' string fields
+    # - 'code.tar.gz' a gzipped tar-fil containing files
+    #    - 'src/...'
+    # as for fpc for now we will package already built artifacts 'enclave.signed.so' and
+    # 'mrenclave', we will skip 'src' and directly place the built artifacts into
+    # the root of 'code.tar.gz'. (Eventually we might add reproducible build to the
+    # external builder, in which case we would stuff the related source into 'src/...'
+    # for metadata.json use the params passed to us, i.e., in particular type 'fpc-c'.
+
+    FPC_PKG_SANDBOX="/tmp/fpc-pkg-sandbox.$$"
+    try mkdir -p ${FPC_PKG_SANDBOX}
+
+    # - create code.tar.gz
+    ENCLAVE_FILE="enclave.signed.so"
+    MRENCLAVE_FILE="mrenclave"
```

Comment:
Do we need this mrenclave file?
I think this is here for backward compatibility -- and it's fine.
As we move on, the mrenclave should be loaded from elsewhere -- likely ercc.

Reply:
The current code needs it functionally afaicr.  But in principle i also don't think we really need it. We have discussed that in the past and punted the related discussion when we refactor to the e2e secure version. So that time is certainly now (but probably not in this PR :-)